File: tayngangstudio_sever/DataAccessLayer/Models/Order.cs
using DataAccessLayer.Enums;
using DataAccessLayer.Models.AbstractEntities;

namespace DataAccessLayer.Models
{
    public class Order : BaseEntity
    {
        public DateTime OrderDate { get; set; } = DateTime.UtcNow;

        public double TotalAmount { get; set; }

        public OrderStatusEnum Status { get; set; } = OrderStatusEnum.Pending;

        public PaymentMethodEnum PaymentMethod { get; set; }

        public int? UserId { get; set; }

        public User User { get; set; }

        public ICollection<OrderDetail> OrderDetails { get; set; } = [];
    }
}

File: tayngang_studio/types/order.ts
export interface OrderDetail {
	quantity: number
	unitPrice: number
	productId: number
	product: {
		name: string
	}
}

export interface Order {
	id: number
	orderDate: string
	totalAmount: number
	status: 'Pending' | 'Canceled' | 'Completed'
	paymentMethod: 'PayOS' | 'Cash'
	paymentUrl: string | null
	userId: number
	orderDetails: OrderDetail[]
}

export interface CreateOrderDto {
	userId: number
	paymentMethod: 'PayOS' | 'Cash'
	orderDetails: {
		quantity: number
		unitPrice: number
		productId: number
	}[]
}
